operator/checkcontroller/executor.go

```diff
@@ -93,3 +108,77 @@ func (c *CheckExecutor) setupEnvVars(ctx context.Context) []corev1.EnvVar {
 func (c *CheckExecutor) cleanupOldChecks(ctx context.Context, check *k8upv1.Check) {
 	c.CleanupOldResources(ctx, &k8upv1.CheckList{}, check.Namespace, check)
 }
+
+func (c *CheckExecutor) appendTLSOptionsArgs() []string {
```

Comment:
You can generalize this function a bit. Instead of making it a method of the *Executor objects you could have something like:
```go
func AppendTLSOptionsArgs(opts *k8upv1.TLSOptions) []string
```

And put it in the new `utils` package. That will reduce a lot of code duplication.

Reply:
Fiexd